src/components/BookDetails.jsx

```diff
@@ -0,0 +1,77 @@
+import React, { useState } from "react";
+import AddBookButton from "./ui/AddBookButton";
+
+
+const BookDetails = (clickedBook) => {
+   const book = clickedBook.clickedBook;
+
+    function processCleanBook(book) {
```

Comment:
Hey, Not sure if you can still see this since this pull request is already pulled. 

A huge amount of the books served by the API are missing various pieces of data and return undefined for some of there values. The processCleanBook is a way to loop through the key value pairs and check for undefined and/or null values. 

Additionally when SpringBoot is processing them, they have to be able to fit into the database. 

Some of the database column values for Book are Varchar(255) while others are Varbinary(255), which I looked up and is actually 340 chars. But if any of the values are longer than these database limits then the POST will fail and the controller throws an error. 

Some of the subject array's have a ton of data. I think Of Mice and Men had 61 indices in the return array just for subject. I first truncated them to just the first 10 indices of the array, but that was still failing. So I landed on 5 as some of them for books are randomly lengthy values.